src/datasets/table.py

```diff
@@ -0,0 +1,591 @@
+import copy
+from functools import wraps
+from typing import List, TypeVar, Union
+
+import pyarrow as pa
+
+
+def wraps_arrow_table_method(arrow_table_method):
+    def wrapper(method):
+        out = wraps(arrow_table_method)(method)
+        out.__doc__ = out.__doc__.replace("pyarrow.Table", "Table")
+        return out
+
+    return wrapper
+
+
+def in_memory_arrow_table_from_file(filename: str) -> pa.Table:
+    in_memory_stream = pa.input_stream(filename)
+    opened_stream = pa.ipc.open_stream(in_memory_stream)
+    pa_table = opened_stream.read_all()
+    return pa_table
+
+
+def in_memory_arrow_table_from_buffer(buffer: pa.Buffer) -> pa.Table:
+    stream = pa.BufferReader(buffer)
+    opened_stream = pa.ipc.open_stream(stream)
+    table = opened_stream.read_all()
+    return table
+
+
+def memory_mapped_arrow_table_from_file(filename: str) -> pa.Table:
```

Comment:
these could be `@classmethod`?

Reply:
Since they return arrow tables and not InMemoryTable or MemoryMapped table, I hesitated to have them as classmethods. No strong opinion though

Reply:
oh yes my bad, misread the `pa.Table` into `Table`. maybe prefix them with an underscore as they seem to be a class util that wouldn't be used from outside?